WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -1087,16 +1083,9 @@ public void onClick(View v) {
                 break;
             case RequestCodes.STORIES_PHOTO_PICKER:
             case RequestCodes.PHOTO_PICKER:
-                Fragment fragment = mBottomNav.getActiveFragment();
-                // TODO move this logic directly to the fragment
-                if (fragment instanceof MySiteFragment) {
-                    fragment.onActivityResult(requestCode, resultCode, data);
-                }
-                break;
+            case RequestCodes.SITE_ICON_PICKER:
```

Comment:
Since we're launching the media picker from `ImprovedMySiteFragment`, wouldn't it be possible to handle this result directly into that fragment instead of having to handle it here first?

Reply:
Good catch, I've fixed it in ac9ab4d